src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -320,23 +321,75 @@ public void removeWorkspaceRole(
   @Traced
   public String syncWorkspacePolicy(
       UUID workspaceId, WsmIamRole role, AuthenticatedUserRequest userReq) {
+    String group =
+        syncPolicyOnObject(
+            SamConstants.SAM_WORKSPACE_RESOURCE, workspaceId.toString(), role.toSamRole(), userReq);
+    logger.info(
+        "Synced role {} to google group {} in workspace {}", role.toSamRole(), group, workspaceId);
+    return group;
+  }
+
+  /**
+   * Wrapper around Sam client to sync a Sam policy on a controlled resource to a google group and
+   * return the email of that group.
+   *
+   * <p>This should only be called for controlled resources which require permissions granted to
+   * individual users, i.e. private or application-controlled resources. All other cases are handled
+   * by the permissions that workspace-level roles inherit on resources, and do not use the policies
+   * synced by this function.
+   *
+   * <p>This operation in Sam is idempotent, so we don't worry about calling this multiple times.
+   *
+   * @param resource The resource to sync a binding for
+   * @param role The policy to sync in Sam
+   * @param userReq User authentication
+   * @return
+   */
+  @Traced
+  public String syncPrivateResourcePolicy(
+      ControlledResource resource,
```

Comment:
little funny that we only need the workspaceId for `syncWorkspacePolicy` but we need the entire ControlledResource class here. I think we only use the resource "category" access/managed by and the resource id.

Reply:
Yeah - I think I'd rather pass the whole object than 3 separate fields, but it's not fantastic.